### ISSUE TITLE
Support ARG/ENV for pip install --extra-index-url 

### ISSUE THREAD

**Issue by isuftin** [hadolint/hadolint#407]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

No errors during hadolint

### Actual behavior

Dockerfile:12 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`

### Steps to reproduce the behavior

Does not work:

```
FROM alpine:3.11.3

ENV MIRROR_ARTIFACTORY=https://my.artifactory.local/artifactory/api/pypi/mirror.files.pythonhosted.org/simple

RUN apk update && apk -v --update add --no-cache \
      python2=2.7.16-r3 \
      py-pip=18.1-r0 \
      groff=1.22.4-r0 \
      less=551-r0 \
      mailcap=2.1.48-r0

RUN pip install --extra-index-url $MIRROR_ARTIFACTORY awscli==1.18.22
```

works:
```
FROM alpine:3.11.3

RUN apk update && apk -v --update add --no-cache \
      python2=2.7.16-r3 \
      py-pip=18.1-r0 \
      groff=1.22.4-r0 \
      less=551-r0 \
      mailcap=2.1.48-r0

RUN pip install --extra-index-url https://my.artifactory.local/artifactory/api/pypi/mirror.files.pythonhosted.org/simple awscli==1.18.22

```
**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
$ hadolint --version
Haskell Dockerfile Linter 1.17.4-no-git
```

**Additional environment details (OS, stack version, etc.)**

MacOS 10.14.6, hadolint installed via homebrew
